wifi-subsys/components/tools/src/serialization.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file   Serialization.h
+ * @author Locha Mesh Developers (contact@locha.io)
+ * @brief
+ * @{
+ */
+
+#include "esp_err.h"
+#include "cbor.h"
+
+#ifndef SERIALIZATION_H
+#define SERIALIZATION_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @brief method to encode data
+ * @param[out] iter CborValue where will be encode the inp_msg
+ * @param buf_size buffer size
```

Comment:
i don't not understand this doc bro,  because saw that Iter es the second parameter, not first and bug_size what? the first parameters seem the data to decode, please be more clear about the parameters.